logger.go

```diff
@@ -14,147 +14,157 @@ type Logger struct {
 }
 
 // NewLogger creates a Logger instance with no additional data.
-func NewLogger() Logger {
-	return Logger{
+func NewLogger() *Logger {
+	return &Logger{
 		loggingT: &logging,
 	}
 }
 
 // WithContext creates a logger from a context.Context
-func WithContext(ctx context.Context) Logger {
+func WithContext(ctx context.Context) *Logger {
 	data, _ := ctx.Value(contextKeyData).([]interface{})
 	prefix, _ := ctx.Value(contextKeyPrefix).(string)
-	return Logger{
+	return &Logger{
 		loggingT: &logging,
 		data:     data,
 		prefix:   prefix,
```

Comment:
I think this may panic if ctx.Value(contextKeyPrefix) is not string

Reply:
Only our code can ever read and write this, so it's safe within this package's context.